gapis/api/vulkan/frame_loop.go

```diff
@@ -33,62 +33,48 @@ type stateWatcher struct {
 	memoryWrites map[memory.PoolID]*interval.U64SpanList
 }
 
-func (b *stateWatcher) OnBeginCmd(ctx context.Context, cmdID api.CmdID, cmd api.Cmd) {
-}
-
-func (b *stateWatcher) OnEndCmd(ctx context.Context, cmdID api.CmdID, cmd api.Cmd) {
-}
+func (b *stateWatcher) OnBeginCmd(ctx context.Context, cmdID api.CmdID, cmd api.Cmd) {}
+func (b *stateWatcher) OnEndCmd(ctx context.Context, cmdID api.CmdID, cmd api.Cmd)   {}
 func (b *stateWatcher) OnBeginSubCmd(ctx context.Context, subIdx api.SubCmdIdx, recordIdx api.RecordIdx) {
 }
-func (b *stateWatcher) OnRecordSubCmd(ctx context.Context, recordIdx api.RecordIdx) {
-}
-func (b *stateWatcher) OnEndSubCmd(ctx context.Context) {
-}
+func (b *stateWatcher) OnRecordSubCmd(ctx context.Context, recordIdx api.RecordIdx) {}
+func (b *stateWatcher) OnEndSubCmd(ctx context.Context)                             {}
 func (b *stateWatcher) OnReadFrag(ctx context.Context, owner api.RefObject, frag api.Fragment, valueRef api.RefObject, track bool) {
 }
-
 func (b *stateWatcher) OnWriteFrag(ctx context.Context, owner api.RefObject, frag api.Fragment, oldValueRef api.RefObject, newValueRef api.RefObject, track bool) {
-
 }
 
 func (b *stateWatcher) OnWriteSlice(ctx context.Context, slice memory.Slice) {
+
 	span := interval.U64Span{
 		Start: slice.Base(),
 		End:   slice.Base() + slice.Size(),
 	}
+
 	poolID := slice.Pool()
+
 	if _, ok := b.memoryWrites[poolID]; !ok {
 		b.memoryWrites[poolID] = &interval.U64SpanList{}
 	}
-	interval.Merge(b.memoryWrites[poolID], span, true)
-}
-
-func (b *stateWatcher) OnReadSlice(ctx context.Context, slice memory.Slice) {
-}
-
-func (b *stateWatcher) OnWriteObs(ctx context.Context, obs []api.CmdObservation) {
-}
 
-func (b *stateWatcher) OnReadObs(ctx context.Context, obs []api.CmdObservation) {
-}
-
-func (b *stateWatcher) OpenForwardDependency(ctx context.Context, dependencyID interface{}) {
-}
-
-func (b *stateWatcher) CloseForwardDependency(ctx context.Context, dependencyID interface{}) {
+	interval.Merge(b.memoryWrites[poolID], span, true)
 }
 
-func (b *stateWatcher) DropForwardDependency(ctx context.Context, dependencyID interface{}) {
-}
+func (b *stateWatcher) OnReadSlice(ctx context.Context, slice memory.Slice)                  {}
+func (b *stateWatcher) OnWriteObs(ctx context.Context, observations []api.CmdObservation)    {}
+func (b *stateWatcher) OnReadObs(ctx context.Context, observations []api.CmdObservation)     {}
+func (b *stateWatcher) OpenForwardDependency(ctx context.Context, dependencyID interface{})  {}
+func (b *stateWatcher) CloseForwardDependency(ctx context.Context, dependencyID interface{}) {}
+func (b *stateWatcher) DropForwardDependency(ctx context.Context, dependencyID interface{})  {}
 
 // Transfrom
 type frameLoop struct {
 	capture        *capture.GraphicsCapture
 	cmds           []api.Cmd
 	numInitialCmds int
 	loopCount      int32
-	loopStartCmd   api.Cmd
-	loopEndCmd     api.Cmd
+	loopStartIdx   int
```

Comment:
I think we want to avoid using indices here. Other transforms may insert commands so indices can get out of scope.